src/Tribe/Attendee_Registration/Template.php

```diff
@@ -123,21 +123,26 @@ public function set_page_template( $template ) {
 			$template = tribe_get_option( 'tribeEventsTemplate', 'default' );
 		}
 
-		switch ( $template ) {
-			case '' :
-			case 'default' :
-				// A bit of logic for themes without a page.php
-				$page = locate_template( 'page.php' );
-				$page = ! empty( $page ) ? 'page.php' : array_keys( wp_get_theme()->get_page_templates() );
-				$page = ! empty( $page ) ? $page : 'index.php';
-				$page = ! is_array( $page ) ? $page : $page[0];
-
-				$template = get_stylesheet_directory() . '/' . $page;
-				break;
-			default :
-				$template = get_stylesheet_directory() . '/' . $template;
+		if ( in_array( $template, array( '', 'default' ), true ) ) {
+			// A bit of logic for themes without a page.php
+			$template = 'page.php;
+
+			if ( ! locate_template( $template ) ) {
+				$pages = array_keys( wp_get_theme()->get_page_templates() );
+
+				if ( ! empty( $pages ) ) {
+					$template = $pages[0];
+				}
+			}
 		}
 
+		// If template is not found, use default.
+		if ( ! locate_template( $template ) ) {
+			$template = 'index.php';
+		}
+
+		$template = get_stylesheet_directory() . '/' . $template;
```

Comment:
I know why this was done (I did it) but maybe we should use `locate_template` here instead? This will break for child themes that don't override the parent template :/

Reply:
Also, if you need a ticket for that part, it's 123613 😉

Reply:
```suggestion
		$template = locate_template( $template );
```